src/espIdf/hints/index.ts

```diff
@@ -0,0 +1,230 @@
+import * as os from "os";
+import * as yaml from "js-yaml";
+import { readFile, pathExists } from "fs-extra";
+import * as idfConf from "../../idfConfiguration";
+import { Logger } from "../../logger/logger";
+import * as utils from "../../utils";
+import * as vscode from "vscode";
+
+class ReHintPair {
+  re: string;
+  hint: string;
+  match_to_output: boolean;
+
+  constructor(re: string, hint: string, match_to_output: boolean = false) {
+    this.re = re;
+    this.hint = hint;
+    this.match_to_output = match_to_output;
+  }
+}
+
+class ErrorHint {
+  public type: "error" | "hint";
+  public children: ErrorHint[] = [];
+
+  constructor(public label: string, type: "error" | "hint") {
+    this.type = type;
+  }
+
+  addChild(child: ErrorHint) {
+    this.children.push(child);
+  }
+}
+
+export class ErrorHintProvider implements vscode.TreeDataProvider<ErrorHint> {
+  constructor(private context: vscode.ExtensionContext) {}
+  private _onDidChangeTreeData: vscode.EventEmitter<
+    ErrorHint | undefined | null | void
+  > = new vscode.EventEmitter<ErrorHint | undefined | null | void>();
+  readonly onDidChangeTreeData: vscode.Event<
+    ErrorHint | undefined | null | void
+  > = this._onDidChangeTreeData.event;
+
+  private data: ErrorHint[] = [];
+
+  async searchError(errorMsg: string, workspace) {
+    const espIdfPath = idfConf.readParameter(
+      "idf.espIdfPath",
+      workspace
+    ) as string;
+    const version = await utils.getEspIdfFromCMake(espIdfPath);
+
+    if (utils.compareVersion(version.trim(), "5.0") === -1) {
```

Comment:
Shouldn't be ESP-IDF v5.0 or higher ? Or is only supported in v5.0 ?

Reply:
It doesn't verify if it's 5.0, it actually verifies if it's higher or not than 5.0